pkg/channel/lookup.go

```diff
@@ -328,3 +329,18 @@ func NewMoreThanOneTemplateCandidateErr(component v1beta2.Module,
 	return fmt.Errorf("%w: more than one module template found for module: %s, candidates: %v",
 		ErrTemplateNotIdentified, component.Name, candidates)
 }
+
+func saveDescriptorToCache(template ModuleTemplateTO) {
+	if template.ModuleTemplate != nil {
+		descriptor := template.GetDescFromCache()
+		if descriptor == nil {
+			var err error
+			descriptor, err = template.GetDescriptor()
+			if err != nil {
```

Comment:
please do not handle `template.Err` in this function, instead, return it, and handle it outside, then you minimum the responsibility of this function to only deal with read and save cache, nothing more.

```
if err := saveDescriptorToCache(); err != nil {
  template.Err = fmt.Errorf("failed to get descriptor: %w", err)
  continue
}
```